### DISCUSSION TITLE
No python version marker on importlib-metadata

### DISCUSSION THREAD

**Discussion by pawamoy** [mkdocs/mkdocs#2956] [General]
The `importlib-metadata` dependency was bumped to `>=4.3` in https://github.com/mkdocs/mkdocs/commit/ec9b583b02376b909c10889fef780cdf2f25a23d.

We're now getting resolution conflicts with Flake8 because they set an upper bound on `importlib-metadata`: `<4.3`. From their answers in different issues and PRs (starting point here: https://github.com/PyCQA/flake8/issues/1503), it seems there has been quite some friction regarding this, and they likely won't change their upper bound.

Back to MkDocs: shouldn't the dependency have a marker like `python_version < '3.8'`?

Actual conflict we get (PDM output, URLs stripped):

```
Conflicts detected: 
  importlib-metadata>=4.3 (from <Candidate mkdocs 1.3.1)
  importlib-metadata; python_version < "3.8" (from <Candidate flake8-comprehensions 3.10.0)
  importlib-metadata; python_version < "3.8" (from <Candidate flake8-tidy-imports 4.8.0)
  importlib-metadata>=0.12; python_version < "3.8" (from <Candidate pytest 7.1.2)
  importlib-metadata>=3.6.0; python_version < "3.10" (from <Candidate pytest-randomly 3.12.0)
  importlib-metadata; python_version < "3.8" (from <Candidate wps-light 0.16.1)
  importlib-metadata<4.3,>=1.1.0; python_version < "3.8" (from <Candidate flake8 5.0.4)
```

We're using Python 3.8 so I believe adding the marker would help in our case at least. Unless the right marker to use is `python_version < '3.10'`.

**Comment by oprypin** [comment]
* Moved to issue and solved: #2958
